pkgs/by-name/wi/winbox4/package.nix

```diff
@@ -0,0 +1,110 @@
+{
+  autoPatchelfHook,
+  copyDesktopItems,
+  fetchurl,
+  fontconfig,
+  freetype,
+  lib,
+  libGL,
+  libxkbcommon,
+  makeDesktopItem,
+  makeWrapper,
+  stdenvNoCC,
+  unzip,
+  writeShellApplication,
+  xorg,
+  zlib,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "winbox";
+  version = "4.0beta6";
+
+  src = fetchurl {
+    name = "WinBox_Linux-${finalAttrs.version}.zip";
+    url = "https://download.mikrotik.com/routeros/winbox/${finalAttrs.version}/WinBox_Linux.zip";
+    hash = "sha256-PCb1jQDzSblYdf+avK/IYQeYgD19Jx8/Kwdg9/EgxNs=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    # makeBinaryWrapper does not support --run
+    makeWrapper
+    unzip
+  ];
+
+  buildInputs = [
+    fontconfig
+    freetype
+    libGL
+    libxkbcommon
+    xorg.libxcb
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
+    xorg.xcbutilwm
+    zlib
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 "assets/img/winbox.png" "$out/share/pixmaps/winbox.png"
+    install -Dm755 "WinBox" "$out/bin/WinBox"
+
+    wrapProgram "$out/bin/WinBox" --run "${lib.getExe finalAttrs.migrationScript}"
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "winbox";
+      desktopName = "Winbox";
+      comment = "GUI administration for Mikrotik RouterOS";
+      exec = "WinBox";
+      icon = "winbox";
+      categories = [ "Utility" ];
+    })
+  ];
+
+  migrationScript = writeShellApplication {
+    name = "winbox-migrate";
+    text = ''
+      XDG_DATA_HOME=''${XDG_DATA_HOME:-$HOME/.local/share}
+      sourceFile="$HOME/.wine/drive_c/users/$USER/Application Data/Mikrotik/Winbox/Addresses.cdb"
+      targetFile="$XDG_DATA_HOME/MikroTik/WinBox/Addresses.cdb"
+
+      if [ ! -f "$sourceFile" ]; then
+        echo "NixOS: WinBox 3 data not found at $(dirname "$sourceFile"). Skipping automatic migration."
+        exit 0
+      fi
+
+      if [ -f "$targetFile" ]; then
+        echo "NixOS: WinBox 4 data already present at $(dirname "$targetFile"). Skipping automatic migration."
+        exit 0
+      fi
+
+      echo "NixOS: Automatically migrating WinBox 3 data..."
+      mkdir -p "$(dirname "$targetFile")"
+      cp "$sourceFile" "$targetFile"
+    '';
+  };
+
+  meta = {
+    description = "Graphical configuration utility for RouterOS-based devices";
+    homepage = "https://mikrotik.com";
+    downloadPage = "https://mikrotik.com/download";
+    changelog = "https://wiki.mikrotik.com/wiki/Winbox_changelog";
```

Comment:
Maybe once winbox 4 goes out of beta this URL will be correct, but right now this is winbox 3.x changelog. `https://download.mikrotik.com/routeros/winbox/4.0beta6/CHANGELOG` has the changelog (but it's always downloaded)

Reply:
I doubt it will ever be correct. The changelog isn't even correct anymore for the v3 and should be changed there as well.

https://github.com/NixOS/nixpkgs/pull/329618/files/3b1674a4110a04b1f063eeede82a0f393a28660d#r1689714775